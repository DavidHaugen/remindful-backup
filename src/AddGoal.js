import React, { Component } from 'react'
import remindfulApiService from './services/remindful-api-service'

export default class AddGoal extends Component {
  submitNewGoal = (e) => {
    e.preventDefault()
    const name = e.target.name.value;
    remindfulApiService.postNewGoal(name)
      .then((goal) => this.props.addGoal(goal))
    this.props.history.push('/my-goals')
  }

  render(){
    return (
      <div>
        <form onSubmit={(e) => this.submitNewGoal(e)}>
          <label>Name your goal:
            <input type='text' id = 'name' ></input>
          </label>
          <button type='submit'>Submit</button>
        </form>
      </div>
    )
  }
}