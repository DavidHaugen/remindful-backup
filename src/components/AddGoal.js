import React, { Component } from 'react'
import remindfulApiService from '../services/remindful-api-service'
import GoalsContext from '../context/GoalsContext'

export default class AddGoal extends Component {

  static contextType = GoalsContext

  submitNewGoal = (e) => {
    e.preventDefault()
    const name = e.target.name.value;
    remindfulApiService.postNewGoal(name)
      .then((goal) => this.context.addGoal(goal))
    this.props.history.push('/my-goals')
  }

  render(){
    return (
      <div className="main">
        <div className="wrapper">
          <h1>New goal!</h1>
          <form className="addGoal" onSubmit={(e) => this.submitNewGoal(e)} >
            <label htmlFor="name" className="addGoalLabel">Name your goal:</label>
            <div className="formContainer">
              <input type='text' id = 'name' required className="addGoalInput" ></input>
            <button type='submit'><i className="fas fa-plus-circle addGoalButton"></i></button>
            </div>
        </form>
        </div>
      </div>
    )
  }
}