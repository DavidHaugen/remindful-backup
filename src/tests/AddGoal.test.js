import React from 'react';
import {MemoryRouter} from 'react-router-dom'
import AddGoal from '../components/AddGoal'
import {mount} from 'enzyme';
import renderer from 'react-test-renderer'

describe('<AddGoal />', () => {
  it('Renders without crashing', () => {
      mount(<MemoryRouter><AddGoal/></MemoryRouter>);
  });
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<MemoryRouter><AddGoal/></MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
    });
});