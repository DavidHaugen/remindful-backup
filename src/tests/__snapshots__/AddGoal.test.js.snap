// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddGoal /> renders the UI as expected 1`] = `
<div
  className="main"
>
  <div
    className="wrapper"
  >
    <h1>
      New goal!
    </h1>
    <form
      className="addGoal"
      onSubmit={[Function]}
    >
      <label
        className="addGoalLabel"
        htmlFor="name"
      >
        Name your goal:
      </label>
      <div
        className="formContainer"
      >
        <input
          className="addGoalInput"
          id="name"
          required={true}
          type="text"
        />
        <button
          type="submit"
        >
          <i
            className="fas fa-plus-circle addGoalButton"
          />
        </button>
      </div>
    </form>
  </div>
</div>
`;
